schema.ts

```diff
@@ -1423,15 +1423,48 @@ export interface AnyonePrincipal {
 type Principal = UserPrincipal | GroupPrincipal | AnyonePrincipal | DomainPrincipal;
 
 /**
- * This represents the definition of a permission in the external system.
+ * The type of permission.
+ * Delegated permissions are permissions that are inherited from another object that aren't
+ * within the same sync table.
+ * Direct permissions are permissions that are directly applied to the object. E.g. a user or group, etc
+ * TODO(drew): Unhide this
+ * @hidden
+ */
+export enum PermissionType {
+  Delegated = 'delegated',
+  Direct = 'direct',
+}
+
+/**
+ * This represents the definition of a direct permission in the external system.
  *
  * TODO(sam): Unhide this
  * @hidden
  */
-export interface Permission {
+export interface DirectPermission {
+  permissionType?: PermissionType.Direct;
   principal: Principal;
 }
 
+/**
+ * This represents the definition of a delegated permission in the external system.
+ * 
+ * TODO(drew): Unhide this
+ * @hidden
+ */
+export interface DelegatedPermission {
+  permissionType: PermissionType.Delegated;
+  delegatedItemId: string | number;
```

Comment:
Can we add the sync table name here we are referencing as well?

Reply:
Would be good to have and would allow more efficient joins in the future potentially.